tests/test_coreg/test_affine.py

```diff
@@ -139,38 +244,49 @@ def test_coreg_example(self, verbose: bool = False) -> None:
         test_examples.py, but helps identify from where differences arise.
         """
 
+        # Use full DEMs here (to compare to original values from older package versions)
+        ref, tba = load_examples(crop=False)[0:2]
+        inlier_mask = ~self.outlines.create_mask(ref)
+
         # Run co-registration
         nuth_kaab = xdem.coreg.NuthKaab()
-        nuth_kaab.fit(self.ref, self.tba, inlier_mask=self.inlier_mask, verbose=verbose, random_state=42)
+        nuth_kaab.fit(ref, tba, inlier_mask=inlier_mask, verbose=verbose, random_state=42)
 
         # Check the output .metadata is always the same
-        shifts = (nuth_kaab.meta["shift_x"], nuth_kaab.meta["shift_y"], nuth_kaab.meta["shift_z"])
-        res = self.ref.res[0]
-        assert shifts == pytest.approx((-0.463 * res, -0.1339999 * res, -1.9922009))
+        shifts = (
+            nuth_kaab.meta["outputs"]["affine"]["shift_x"],
+            nuth_kaab.meta["outputs"]["affine"]["shift_y"],
+            nuth_kaab.meta["outputs"]["affine"]["shift_z"],
+        )
+        assert shifts == pytest.approx((-9.200801, -2.785496, -1.9818556))
 
-    def test_gradientdescending(self, subsample: int = 10000, inlier_mask: bool = True, verbose: bool = False) -> None:
+    def test_gradientdescending(self, subsample: int = 10000, verbose: bool = False) -> None:
         """
         Test the co-registration outputs performed on the example are always the same. This overlaps with the test in
         test_examples.py, but helps identify from where differences arise.
 
         It also implicitly tests the z_name kwarg and whether a geometry column can be provided instead of E/N cols.
         """
-        if inlier_mask:
-            inlier_mask = self.inlier_mask
+        # Use full DEMs here (to compare to original values from older package versions)
+        ref, tba = load_examples(crop=False)[0:2]
+        inlier_mask = ~self.outlines.create_mask(ref)
 
         # Run co-registration
         gds = xdem.coreg.GradientDescending(subsample=subsample)
         gds.fit(
-            self.ref.to_pointcloud(data_column_name="z").ds,
-            self.tba,
+            ref.to_pointcloud(data_column_name="z").ds,
+            tba,
             inlier_mask=inlier_mask,
             verbose=verbose,
             random_state=42,
         )
 
-        res = self.ref.res[0]
-        shifts = (gds.meta["shift_x"], gds.meta["shift_y"], gds.meta["shift_z"])
-        assert shifts == pytest.approx((0.03525 * res, -0.59775 * res, -2.39144), abs=10e-5)
+        shifts = (
+            gds.meta["outputs"]["affine"]["shift_x"],
+            gds.meta["outputs"]["affine"]["shift_y"],
+            gds.meta["outputs"]["affine"]["shift_z"],
+        )
+        assert shifts == pytest.approx((-10.625, -2.65625, 1.940031), abs=10e-5)
```

Comment:
why are the shifts completely different compared to the old tests?

Reply:
There's a half pixel shift (10 meters) because the `GradientDescending` used to force a pixel interpretation that was not the one in the DEM.
Now the shifts match that of `NuthKaab`, so it's more consistent!